### PR TITLE
cmd: enable chaindatafetcher feature to kpn and kspn

### PR DESCRIPTION
**Pull request by ehnuje**
## Proposed changes

- This PR enables kpn and kspn to use chaindatafetcher feature
- ChainDataFetcherFlags is introduced to avoid redundant codes

## Types of changes

Please put an x in the boxes related to your change.

- [ ] Bugfix
- [x] New feature or enhancement
- [ ] Others

## Checklist

*Put an x in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code.*

- [x] I have read the [CONTRIBUTING GUIDELINES](https://github.com/klaytn/klaytn/blob/master/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://cla-assistant.io/klaytn/klaytn)
- [x] Lint and unit tests pass locally with my changes (`$ make test`)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Related issues

- Please leave the issue numbers or links related to this PR here.

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Review by jeongkyun-oh — DISMISSED** [review]

**Review by jeongkyun-oh — APPROVED** [review]

**Review by hqjang-pepper — COMMENTED** [review]
Is DBmigration option used in both EN and PN, not in CN?

**Comment by ehnuje** [comment]
> Is DBmigration option used in both EN and PN, not in CN?

@hqjang-pepper If you're talking about `DBMigrationFlags`, it is used by CN, PN and EN, however not by service chain nodes.

**Comment by hqjang-pepper** [comment]
Oh sorry I took a wrong look.

**Review by hqjang-pepper — APPROVED** [review]

**Comment by ehnuje** [comment]
@aidan-kwon @henry-will @2dvorak @nohkwak Dear code owners, could you please take a look?

**Review by 2dvorak — DISMISSED** [review]

**Review by zacscoding — APPROVED** [review]
@ehnuje LGTM~ I tested kafka with KPN :)

**Review by hqjang-pepper — APPROVED** [review]

**Review by jeongkyun-oh — APPROVED** [review]

**Comment by ehnuje** [comment]
@aidan-kwon @henry-will @2dvorak @nohkwak Dear code owners, could you please take a look?

File: cmd/utils/nodecmd/nodeflags.go [hunk 1]
```diff
@@ -452,3 +402,31 @@ var DBMigrationFlags = []cli.Flag{
...
 	altsrc.NewInt64Flag(utils.DstDynamoDBWriteCapacityFlag),
 }
+
+var ChainDataFetcherFlags = []cli.Flag{
+	altsrc.NewBoolFlag(utils.EnableChainDataFetcherFlag),
+	altsrc.NewStringFlag(utils.ChainDataFetcherMode),
+	altsrc.NewBoolFlag(utils.ChainDataFetcherNoDefault),
+	altsrc.NewIntFlag(utils.ChainDataFetcherNumHandlers),
+	altsrc.NewIntFlag(utils.ChainDataFetcherJobChannelSize),
+	altsrc.NewIntFlag(utils.ChainDataFetcherChainEventSizeFlag),
```
Comment (aidan-kwon) [inline cmd/utils/nodecmd/nodeflags.go:412]:
```suggestion
	altsrc.NewIntFlag(utils.ChainDataFetcherChainEventSizeFlag),
	altsrc.NewIntFlag(utils.ChainDataFetcherMaxProcessingDataSize),
```

**Review by aidan-kwon — COMMENTED** [review]
Why don't you add those flags to CN/SCN?

**Comment by ehnuje** [comment]
> Why don't you add those flags to CN/SCN?

@aidan-kwon According to [this comment](https://github.com/klaytn/klaytn/issues/1341#issuecomment-1120870904), it needs additional modifications to use ChainDataFetcher in (S)CN.

File: cmd/utils/nodecmd/nodeflags.go (hunk 1 above)
Comment (ehnuje) [reply cmd/utils/nodecmd/nodeflags.go:412]: Hawk eye! 🙏

**Review by henry-will — APPROVED** [review]
Good job about duplicate flag codes. I checked in ChainDataFetcherFlags to see if it was the same as previous flag list. 
I heard about that CN and SCN had some problem, but zac checked no issue in PN and SPN.

**Review by jeongkyun-oh — APPROVED** [review]

**Pull request merged**